api/admin/health/health.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package health
+
+import (
+	"sync"
+	"time"
+
+	"github.com/vechain/thor/v2/chain"
+	"github.com/vechain/thor/v2/comm"
+	"github.com/vechain/thor/v2/thor"
+)
+
+type BlockIngestion struct {
+	ID        *thor.Bytes32 `json:"id"`
+	Timestamp *time.Time    `json:"timestamp"`
+}
+
+type Status struct {
+	Healthy           bool            `json:"healthy"`
+	BlockIngestion    *BlockIngestion `json:"blockIngestion"`
+	ChainBootstrapped bool            `json:"chainBootstrapped"`
+	PeerCount         int             `json:"peerCount"`
+}
+
+type Health struct {
+	lock               sync.RWMutex
+	repo               *chain.Repository
+	p2p                *comm.Communicator
+	isNodeBootstrapped bool
+}
+
+const (
+	defaultMaxTimeBetweenSlots = time.Duration(2*thor.BlockInterval) * time.Second
+	defaultMinPeerCount        = 2
+)
+
+func New(repo *chain.Repository, p2p *comm.Communicator) *Health {
+	return &Health{
+		repo: repo,
+		p2p:  p2p,
+	}
+}
+
+// isNetworkProgressing checks if the network is producing new blocks within the allowed interval.
+func (h *Health) isNetworkProgressing(now time.Time, bestBlockTimestamp time.Time, maxTimeBetweenSlots time.Duration) bool {
+	return now.Sub(bestBlockTimestamp) <= maxTimeBetweenSlots
```

Comment:
Does this cater for non-UTC timezones?

Reply:
What leads you to say that? It should consider, as it's a `time.Time` instance right ?

Reply:
```
bestBlockTimestamp := time.Unix(int64(bestBlock.Header.Timestamp()), 0)
```

`bestBlock.Header.Timestamp()` will be the same value, where the node is in America or Asia, because the block timestamp is always UTC. `time.Now()` will differ, depending on whether the node is in America or Asia

Reply:
Ah, I see what you mean.
The `time.Time` holds the timestamp and a location (which I believe it defaults to UTC), so when you're comparing time objects it take into account the location so to avoid timezone issues.

From the https://go.dev/src/time/time.go
```
// Each time has an associated [Location]. The methods [Time.Local], [Time.UTC], and Time.In return a
// Time with a specific Location. Changing the Location of a Time value with
// these methods does not change the actual instant it represents, only the time
// zone in which to interpret it.
```

In this case I think I see an issue because `bestBlockTimestamp` should have a default location of UTC which is correct. So when comparing `timeNowInAsia` with `bestBlockTimestamp` it's going to correctly compare because of it's underlying storage structure.

Reply:
But the block does not store a time.Time, it stores seconds. 

This does not specify the timezone, so thats why I'm confused:

```
time.Unix(int64(bestBlock.Header.Timestamp()), 0)
```

Reply:
Ahh, I got it, `Unix` is always UTC